app/controllers/users/sessions_controller.rb

```diff
@@ -50,5 +63,11 @@ def track_authentication_attempt(email)
 
       analytics.track_event('Authentication Attempt with nonexistent user')
     end
+
+    def password
+      params.require(:user)[:password]
+    rescue ActionController::ParameterMissing
+      ''
```

Comment:
By suppressing the error, it would also prevent us from knowing that our form is broken. The whole point of `params.require` is that we want to enforce the presence of a params by having the app signal to us that the params is missing. If we rescue the error, we are saying it's OK if the params is missing.